src/admin/dumb_components/custom_email_form.js

```diff
@@ -1,19 +1,22 @@
 const React = require('react')
-const { map, objOf, zipWith, merge, compose } = require('ramda')
+const { map, objOf, zipWith, merge, compose, indexBy, prop, converge, dissoc } = require('ramda')
 
 export default ({ members, submit }) => {
   const onSubmit = (e) => {
     e.preventDefault()
-    const template = format_message(e.target[0].value)
-    const emailBodies = map(compose(objOf('email_body'), template), members)
-    const emails = zipWith(merge, members, emailBodies)
-    submit(emails)
+    const template = format_message(e.target)
+    const emailBodies = map(compose(objOf('content'), template), members)
+    const emailsArr = zipWith(merge, members, emailBodies)
+    const setEmailKey = map(compose(indexBy, prop), ['primary_email', 'secondary_email'])
```

Comment:
So I have never called an action (in this case submit_email) from the
reducer. Is that ok?  If we move it all up to the reducer we need to fire 2
actions in total compared to 1.

Not really a problem but wanted to check that is what you meant.

On Thu, Jul 14, 2016 at 12:06 PM, Rafe <notifications@github.com> wrote:

> In src/admin/dumb_components/custom_email_form.js
> <https://github.com/foundersandcoders/sail-back/pull/286#discussion_r70787034>
> :
>
> >
> >  export default ({ members, submit }) => {
> >    const onSubmit = (e) => {
> >      e.preventDefault()
> > -    const template = format_message(e.target[0].value)
> > -    const emailBodies = map(compose(objOf('email_body'), template), members)
> > -    const emails = zipWith(merge, members, emailBodies)
> > -    submit(emails)
> > +    const template = format_message(e.target)
> > +    const emailBodies = map(compose(objOf('content'), template), members)
> > +    const emailsArr = zipWith(merge, members, emailBodies)
> > +    const setEmailKey = map(compose(indexBy, prop), ['primary_email', 'secondary_email'])
>
> So basically the difference is whether format_message gets called in the
> reducer or in the component? Ideally it would be called in the reducer. It
> sounds like you foresee a problem with that though?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/foundersandcoders/sail-back/pull/286/files/02e081a100a86bfc84019925babe1478a6dd75db#r70787034>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AMDYNWog8uodLS0h00AgpkLz-tPOkdaBks5qVhhLgaJpZM4JKksx>
> .
>